src/interfaces/qwrendererinterface.cpp

```diff
@@ -0,0 +1,339 @@
+{
+    auto _handle = static_cast<_wlr_renderer *>(handle);
+    Q_ASSERT(_handle->interface && _handle->impl == handle->impl);
+    return static_cast<_wlr_renderer *>(handle)->interface;
+}
+
+namespace impl {
+static bool bind_buffer(wlr_renderer *handle, wlr_buffer *buffer)
+{
+    return interface(handle)->bindBuffer(QWBuffer::from(buffer));
+}
+
+static void begin(wlr_renderer *handle, uint32_t width, uint32_t height)
+{
+    return interface(handle)->begin({static_cast<int>(width), static_cast<int>(height)});
+}
+
+static void end(wlr_renderer *handle)
+{
+    return interface(handle)->end();
+}
+
+static void clear(wlr_renderer *handle, const float color[4])
+{
+    return interface(handle)->clear(QColor(color[0], color[1], color[2], color[3]));
+}
+
+static void scissor(wlr_renderer *handle, wlr_box *box)
+{
+    return interface(handle)->scissor({box->x, box->y, box->width, box->height});
+}
+
+static bool render_subtexture_with_matrix(wlr_renderer *handle,
+                                          wlr_texture *texture,
+                                          const wlr_fbox *box,
+                                          const float matrix[9],
+                                          float alpha)
+{
+    return interface(handle)->renderSubtextureWithMatrix(QWTexture::from(texture),
+                                                         {box->x, box->y, box->width, box->height},
+                                                         QMatrix3x3(matrix),
+                                                         alpha);
+}
+
+static void render_quad_with_matrix(wlr_renderer *handle,
+                                    const float color[4],
+                                    const float matrix[9])
+{
+    return interface(handle)->renderQuadWithMatrix(QColor(color[0], color[1], color[2], color[3]),
+                                                   QMatrix3x3(matrix));
+}
+
+static const uint32_t *get_shm_texture_formats(wlr_renderer *handle, size_t *len)
+{
+    return interface(handle)->getShmTextureFormats(len).data();
+}
+
+static const wlr_drm_format_set *get_dmabuf_texture_formats(wlr_renderer *handle)
+{
+    return interface(handle)->getDmabufTextureFormats();
+}
+
+static const wlr_drm_format_set *get_render_formats(wlr_renderer *handle)
+{
+    return interface(handle)->getRenderFormats();
+}
+
+static uint32_t preferred_read_format(wlr_renderer *handle)
+{
+    return interface(handle)->preferredReadFormat();
+}
+
+static bool read_pixels(wlr_renderer *handle,
+                        uint32_t fmt,
+                        uint32_t stride,
+                        uint32_t width,
+                        uint32_t height,
+                        uint32_t srcX,
+                        uint32_t srcY,
+                        uint32_t dstX,
+                        uint32_t dstY,
+                        void *data)
+{
+    Q_ASSERT_X(width <= INT_MAX, "QWRendererInterface", "Image width is greater than INT_MAX.");
```

Comment:
Add a function for uint32_t of pixel value convert to int, in the function to assert  the source value is <= INT_MAX. And using the function to replace all `static_cast<int>(...)`.